src/interface/state_descriptor.hpp

```
@@ -148,21 +216,41 @@ class StateDescriptor {
   }
 
   // retrieve metadata for a specific field
-  Metadata &FieldMetadata(const std::string &field_name) {
-    return metadataMap_[field_name];
+  const Metadata &FieldMetadata(const std::string &base_name,
+                                int sparse_id = InvalidSparseID) const {
+    static const Metadata empty_metadata;
+
+    const auto itr = metadataMap_.find(VarID(base_name, sparse_id));
+    if (itr == metadataMap_.end()) {
+      return empty_metadata;
+    } else {
+      return itr->second;
+    }
```

Hmm this is interesting... is an empty metadata object the fail case we want here? Maybe we should just throw an error for a non-existent field?

---

Fine with me. I tried to preserve the old behavior, but I like throwing better.

---

Maybe there's a reason empty metadata was returned but I can't think of one. @jdolence @nmsriram what was the reasoning here? If there wasn't a good reason we should definitely throw.